### PR TITLE
Added 42Crunch parser

### PR DESCRIPTION
**Pull request by FelixHernandez**
## Added 42Crunch parser
[sc-2729]

**Description**

Added 42Crunch parser, docs, and unittests using `DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL` as dedupe field

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

File: docs/content/en/integrations/parsers/file/crunch42.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Crunch42 Scan"
+toc_hide: true
+---
+Import JSON findings from Crunch42 vulnerability scan tool.
```
Comment (manuel-sommer) [inline docs/content/en/integrations/parsers/file/crunch42.md:5]: `Import JSON findings from [Crunch42](https://42crunch.com) vulnerability scan tool.`

Comment (manuel-sommer) [reply docs/content/en/integrations/parsers/file/crunch42.md:5]: Maybe you can get a little more specific regarding which output is provided as there are multiple offers from Crunch42

File: unittests/tools/test_crunch42_parser.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(8, len(findings))
+
+    def test_crunch42parser_single_has_many_findings2(self):
+        testfile = open("unittests/scans/crunch42/crunch42_many_findings2.json")
+        parser = Crunch42Parser()
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(5, len(findings))
```
Comment (Maffooch) [inline unittests/tools/test_crunch42_parser.py:20]: Please add some field validation to you tests as well to make sure everything is being parsed and set on the finding as expected

Here is an example
https://github.com/DefectDojo/django-DefectDojo/blob/82181253cd303efd1642994327e1d862c6b247ef/unittests/tools/test_acunetix360_parser.py#L9C1-L28C131

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: unittests/tools/test_crunch42_parser.py (hunk 2 above)
Comment (FelixHernandez) [reply unittests/tools/test_crunch42_parser.py]: added in https://github.com/DefectDojo/django-DefectDojo/pull/9689/commits/003b713b7d39b489386728d32743c36f624488b7

**Review by grendel513 — APPROVED** [review]

File: dojo/tools/crunch42/parser.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            severity = "High"
+        else:
+            severity = "Critical"
+        # create the finding object
+        finding = Finding(
+            unique_id_from_tool=fiingerprint,
+            title=title,
+            test=test,
+            severity=severity,
+            description="**fingerprint**: " + str(fiingerprint) + "\n"
```
Comment (cneill) [inline dojo/tools/crunch42/parser.py:78]:
```suggestion
            description="**fingerprint**: " + str(fingerprint) + "\n"
```

File: dojo/tools/crunch42/parser.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                for issue in node["issues"]:
+                    item = self.get_item(
+                        issue, key, test
+                    )
+                    items[iterator] = item
+                    iterator += 1
+        return list(items.values())
+
+    def get_item(self, issue, title, test):
+        fiingerprint = issue["fingerprint"]
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/crunch42/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9689/commits/f03801a8667c8dedfd2901273337792081d97a9d

File: dojo/tools/crunch42/parser.py (hunk 3 above)
Comment (FelixHernandez) [reply dojo/tools/crunch42/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9689/commits/f03801a8667c8dedfd2901273337792081d97a9d

File: dojo/tools/crunch42/parser.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            severity = "Low"
+        elif criticality == 3:
+            severity = "Medium"
+        elif criticality <= 4:
+            severity = "High"
+        else:
+            severity = "Critical"
+        # create the finding object
+        finding = Finding(
+            unique_id_from_tool=fiingerprint,
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/crunch42/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9689/commits/f03801a8667c8dedfd2901273337792081d97a9d

**Review by cneill — APPROVED** [review]

**Pull request closed without merging**